### PR TITLE
refactor: add `BlockWriter` and `BlockExecutionWriter` 

### PR DESCRIPTION
**Pull request by joshieDo**
* add `BlockExecutionWriter` 
* add `BlockWriter`
* removes  `utils` mod, alongside its moved functions `insert_block` and `insert_canonical_block`
* removes some `tx_mut`, `&mut Tx` and `mut provider` that got unblocked (and left-over such as `trie.rs` tests).

### CODE REVIEW

File: crates/storage/provider/src/traits/block.rs
Comment (onbjerg) [inline crates/storage/provider/src/traits/block.rs:201]: Ideally this would be an extension trait of `BlockReader + BlockWriter` given it reads *or* writes (takes)

**Review by onbjerg — COMMENTED** [review]
small comment

**Review by onbjerg — APPROVED** [review]
lgtm, pending one more approval

**Review by Rjected — APPROVED** [review]
lgtm, will the other methods be moved to new traits (or these ones) as well?
e.g.
 - `get_take_block_execution_result_range`
 - `get_take_block_transaction_range`
 - etc

**Comment by joshieDo** [comment]
I'm still unsure , because they're private that are only used to help implement the trait methods

**Pull request merged**
